File: algorithm/leetcode5433.the-kth-factor-of-n.cpp
class Solution {
public:
    int kthFactor(int n, int k) {
        set<int> s;
        for(int i = 1; i <= n; i++) {
            if(n%i==0)
                s.insert(i);
        }
        if(s.size() < k)
            return -1;
        auto it = s.begin();
        while(--k && it != s.end())
        {
            it++;
        }
        return *it;
    }
};

File: algorithm/Python3_version/leetcode5433.the-kth-factor-of-n.py
class Solution:
    def kthFactor(self, n: int, k: int) -> int:
        s = set()
        for i in range(1,n+1):
            if n%i == 0:
                s.add(i)
        if len(s) < k:
            return -1
        l = list(s)
        l.sort()
        return l[k-1]